dash/backend/src/modules/falco/service/falco.service.ts

```diff
@@ -120,6 +125,126 @@ export class FalcoService {
+        const falcoId = newFalcoLog[0].id;
+        const falcoSignature = newFalcoLog[0].anomaly_signature;
+        const falcoNamespace = newFalcoLog[0].namespace;
+        const falcoSeverity = newFalcoLog[0].level;
+
+        // Parse data from falco settings json fields
+        const severityLevel = falcoSetting["severity_level"];
+        const anomalyFrequency = (falcoSetting['anomaly_notification_frequency'] * 24 * 60 * 60 * 1000);
+        const weekDay = falcoSetting['weekday'];
+        const emailList = falcoSetting['email_list'].split(',');
+        const cleanEmailList = [];
+        for (const ele of emailList) {
+            cleanEmailList.push(ele.trim());
+        }
+        const whoToNotify = falcoSetting['who_to_notify'];
+
+
+        if (!severityLevel.includes(falcoSeverity)) {
+            return; // no need to send email - does not match the severity level we send emails for
+        }
+
+        // Have we already sent an email ?
+        const lastEmailSentTime = await this.falcoEmailAlreadySent(clusterId, falcoSignature);
+
+        // if no email record then send an email now
+        if (lastEmailSentTime === null || lastEmailSentTime <= (Date.now() - lastEmailSentTime)) {
+            await this.sendFalcoEmailAndAddFalcoEmailRecord(whoToNotify, allAdminEmailArray, emailList, clusterId, falcoId, falcoSeverity, falcoNamespace, falcoSignature, newFalcoLog);
+        } else {
+            // when was the last email sent?
+            const timeDifference = Date.now() - lastEmailSentTime;
+
+            // if the last sent email was older than anomalyFrequency then send
+            if (timeDifference > anomalyFrequency) {
+                await this.sendFalcoEmailAndAddFalcoEmailRecord(whoToNotify, allAdminEmailArray, emailList, clusterId, falcoId, falcoSeverity, falcoNamespace, falcoSignature, newFalcoLog);
+            } else {
+                console.log('Email notification was already sent for this anomaly!');
+            }
+        }
+    }
+
+    async sendFalcoEmailAndAddFalcoEmailRecord(whoToNotify, allAdminEmailArray: any[], emailList, clusterId: number, falcoId, falcoSeverity, falcoNamespace, falcoSignature, newFalcoLog: FalcoDto) {
+        // send to all admin OR specific email list?
+        const usersToSendTo = (whoToNotify === 'allAdmin') ? allAdminEmailArray : emailList;
+
+        for (const user of usersToSendTo) {
+            const emailSentTime = await this.sendFalcoEmail(
+                user,
+                clusterId,
+                falcoId,
+                falcoSeverity,
+                falcoNamespace,
+                falcoSignature,
+                newFalcoLog
+            );
+
+            // if email is sent then add a falco email record
+            if (emailSentTime !== null && emailSentTime !== undefined) {
+                await this.addFalcoEmail(emailSentTime, clusterId, falcoSignature);
+            } else {
+                console.log("Failed to send email!")
```

Comment:
I believe instead of console.log we have a logger class that should be used. 

https://github.com/m9sweeper/m9sweeper/blob/6a09d069725d3381c79b01d3f3d14b1073b152c7/dash/backend/src/modules/command-line/services/kubernetes-cluster.service.ts#L69